more obvious message when wrong config pattern given.

```

sr3 start sender/sender_hpfx-collab-wxo-dd-direct (should have been sender/hpfx-collab-wxo-dd-direct)
starting:( 0 ) Done

```
So it got the list of configs from the command line, and they did not match any existing flow configurations, so it says it started 0 things... technically accurate... but kind of uninformative at the same time.

It would be a lot more useful if it gave some kind error feedback to say it did not find any matching configs.

things discussed:
* if you get zero good configs... obviously do nothing.
* if you have multiple patterns, and some do not match anything... what do you do? 
* I guess fail the whole thing.  thou shalt have correct configs.  It's better to do entire operation or none of it... half is more confusing in practice.
* 


```

SSC-5CD2310S60% sr3 enable hoho/haha
2024-08-09 15:07:32,300 90380 [ERROR] root enable ['hoho/haha'] configuration not found

SSC-5CD2310S60% sr3 disable hoho/haha
2024-08-09 15:07:37,644 90406 [ERROR] root disable ['hoho/haha'] configuration not found

SSC-5CD2310S60% sr3 disable subscribe/u_sftp_f60 hoho/haha
2024-08-09 15:07:49,665 90508 [ERROR] root disable ['hoho/haha'] configuration not found

SSC-5CD2310S60%
SSC-5CD2310S60%
SSC-5CD2310S60% sr3 start subscribe/u_sftp_f60 hoho/haha
starting:2024-08-09 15:08:59,716 90848 [ERROR] root start ['hoho/haha'] configuration not found

SSC-5CD2310S60% sr3 start hoho/haha
starting:2024-08-09 15:09:03,884 90867 [ERROR] root start ['hoho/haha'] configuration not found

SSC-5CD2310S60% sr3 disable subscribe/ftp_f70 subscribe/u_sftp_f60 flow/lovely
2024-08-09 15:10:07,936 91184 [ERROR] root disable ['flow/lovely'] configuration not found

SSC-5CD2310S60% sr3 disable subscribe/ftp_f70 subscribe/u_sftp_f60
2024-08-09 15:10:12,927 91206 [INFO] root disable subscribe/ftp_f70
2024-08-09 15:10:12,927 91206 [INFO] root disable subscribe/u_sftp_f60

SSC-5CD2310S60%

```

```

sr3 --dangerWillRobinson 25 remove cpost/pelle_dd1_f04.conf cpost/pelle_dd2_f05.conf cpost/veille_f34.conf cpump/xvan_f14.conf cpump/xvan_f15.conf poll/sftp_f62.conf poll/sftp_f63.conf post/shim_f63.conf post/t_dd1_f00.conf post/t_dd2_f00.conf post/test2_f61.conf report/tsarra_f20.conf sarra/download_f20.conf sender/tsource2send_f50.conf shovel/rabbitmqtt_f22.conf subscribe/amqp_f30.conf subscribe/cdnld_f21.conf subscribe/cfile_f44.conf subscribe/cp_f61.conf subscribe/ftp_f70.conf subscribe/mirror_f80.conf subscribe/q_f71.conf subscribe/rabbitmqtt_f31.conf subscribe/u_sftp_f60.conf watch/f40.conf cpump/cno_trouble_f00.inc shovel/no_trouble_f00.inc
2024-08-09 15:26:21,313 96404 [ERROR] root cleanup ['cpump/cno_trouble_f00.inc', 'shovel/no_trouble_f00.inc'] configuration not found

```
hmm.... include files...

--- a/sarracenia/config.py
+++ b/sarracenia/config.py
@@ -2055,7 +2055,7 @@ class Config:
                     self.sleep=1
 
         if self.runStateThreshold_hung < self.housekeeping:
-            logger.warning( f"{component}/{config} runStateThreshold_hung {self.runStateThreshold_hung} set lower than housekeeping {self.housekeeping}. sr3 sanity might think this flow is hung kill it too quickly.")
+            logger.warning( f"{component}/{config} runStateThreshold_hung {self.runStateThreshold_hung} set lower than housekeeping {self.housekeeping}. sr3 sanity might think this flow is hung and kill it too quickly.")
 
         if self.vip and not features['vip']['present']:
             logger.critical( f"{component}/{config} vip feature requested, but missing library: {' '.join(features['vip']['modules_needed'])} " )

--- a/sarracenia/sr.py
+++ b/sarracenia/sr.py
@@ -1601,8 +1601,8 @@ class sr_GlobalState:
                 flow=None
 
     def disable(self):
-        if len(self.filtered_configurations) == 0:
-            logging.error('%s configuration not found', self.leftovers)
+        if len(self.leftovers) > 0 and not self._action_all_configs:
+            logging.error( f"{self.leftovers} configuration not found" )
             return
 
         for f in self.filtered_configurations:
@@ -1667,8 +1667,8 @@ class sr_GlobalState:
                 self.run_command(['tail', '-f', lfn])
 
     def enable(self):
-        if len(self.filtered_configurations) == 0:
-            logging.error('%s configuration not found', self.leftovers)
+        if len(self.leftovers) > 0 and not self._action_all_configs:
+            logging.error( f'{self.leftovers} configuration not found' )
             return
         # declare exchanges first.
         for f in self.filtered_configurations:
@@ -1790,6 +1790,10 @@ class sr_GlobalState:
                 print('foreground: stop other instances of this process first')
 
     def cleanup(self) -> bool:
+
+        if len(self.leftovers) > 0 and not self._action_all_configs:
+            logging.error( f'{self.leftovers} configuration not found' )
+            return
 
         if len(self.filtered_configurations) > 1 :
             if len(self.filtered_configurations) != self.options.dangerWillRobinson:
@@ -2065,8 +2069,8 @@ class sr_GlobalState:
 
     def remove(self):
 
-        if len(self.filtered_configurations) == 0:
-            logging.error("No configuration matched")
+        if len(self.leftovers) > 0 and not self._action_all_configs:
+            logging.error( f"{self.leftovers} configuration not found" )
             return
 
         if len(self.filtered_configurations) > 1 :
@@ -2153,6 +2157,10 @@ class sr_GlobalState:
 
         :return:
         """
+        if len(self.leftovers) > 0 and not self._action_all_configs:
+            logging.error( f"{self.leftovers} configuration not found" )
+            return
+
         pcount = 0
         kill_hung=[]
         for f in self.filtered_configurations:
@@ -2236,6 +2244,10 @@ class sr_GlobalState:
         :return:
         """
 
+        if len(self.leftovers) > 0 and not self._action_all_configs:
+            logging.error( f"{self.leftovers} configuration not found" )
+            return
+
         pcount = 0
         for f in self.filtered_configurations:
 
@@ -2282,12 +2294,15 @@ class sr_GlobalState:
            return 0 on success, non-zero on failure.
         """
 
+        if len(self.leftovers) > 0 and not self._action_all_configs:
+            logging.error( f"{self.leftovers} configuration not found" )
+            return
+
         self._clean_missing_proc_state()
 
         if len(self.procs) == 0:
             print('no procs running...already stopped')
             return
-
 
         print('sending SIGTERM ', end='', flush=True)
         pcount = 0
@@ -2535,6 +2550,10 @@ class sr_GlobalState:
     def status(self):
         """ v3 Printing prettier statuses for each component/configs found
         """
+
+        if len(self.leftovers) > 0 and not self._action_all_configs:
+            logging.error( f"{self.leftovers} configuration not found" )
+            return
 
         flowNameWidth=self.cumulative_stats['flowNameWidth']
         latestTransferWidth=self.cumulative_stats['latestTransferWidth']
@@ -2879,6 +2898,11 @@ class sr_GlobalState:
 
         :return:
         """
+
+        if len(self.leftovers) > 0 and not self._action_all_configs:
+            logging.error( f"{self.leftovers} configuration not found" )
+            return
+
         bad = 0
 
 

